client-js/src/client/backend-client.js

```diff
@@ -20,6 +20,94 @@
 
 "use strict";
 
+import {Observable} from './observable';
+import {TypedMessage, TypeUrl} from './typed-message';
+import {QUERY_STRATEGY} from './http-endpoint';
+
+/**
+ * Fetches the results of the query from the server using the provided backend.
+ *
+ * Fetch is a static member of the `BackendClient`.
+ */
+class Fetch {
+
+  /**
+   * @param {!TypedMessage<Query>} query a query to be performed by Spine
+   * @param {!BackendClient} backend the backend which is used to fetch the query results
+   */
+  constructor({of: query, using: backend}) {
+    this._query = query;
+    this._backend = backend;
+  }
+
+  /**
+   * Fetches items one-by-one using an Observable.
+   * Suitable for big collections.
+   *
+   * @returns {Observable<Object, EndpointError>} an Observable retrieving values one at a time.
+   * @example
+   * fetchAll({ofType: taskType}).oneByOne().subscribe({
+   *   next(value) { ... },
+   *   error(error) { ... },
+   *   complete() { ... }
+   * })
+   */
+  oneByOne() {
+    return this._fetchManyOneByOne();
+  }
+
+  /**
+   * Fetches all query results at once resolving a promise with an array of entities.
+   *
+   * @returns {Promise<Object[]>} a Promise resolving an array of items matching query, 
+   *                              that can catch an `EndpointError` 
```

Comment:
- > https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise#Description

  _"A pending promise can either be fulfilled with a value, or rejected with a reason (error). "_

  Seems that it is more correct to say that a `Promise` is "fulled with an array" (not "resolving an array"), and "rejected with `EndpointError` (instead of that it "can catch an (error)").

- _"items"_ -> consider using _"entities"_ for formal correctness and less ambiguity.